packages/zoe/src/zoeService/offer/offer.js

```diff
@@ -36,17 +39,33 @@ export const makeOffer = (
     offerArgs = undefined,
     feePurse,
   ) => {
-    const { instanceHandle, invitationHandle } = await burnInvitation(
-      invitationIssuer,
-      invitation,
-    );
+    const {
+      instanceHandle,
+      invitationHandle,
+      fee,
+      expiry,
+    } = await burnInvitation(invitationIssuer, invitation);
     // AWAIT ///
 
-    await chargeZoeFee(feePurse, offerFeeAmount);
-
     const instanceAdmin = getInstanceAdmin(instanceHandle);
     instanceAdmin.assertAcceptingOffers();
 
+    if (timeAuthority && expiry && fee) {
+      // TODO: is there a way to make this a top-level await?
```

Comment:
Ok I tried and failed to rewrite without significant loss in readability. Thus, no change suggested for this PR. But let's remember to come back to this one.